zerver/tests/tests.py

```diff
@@ -542,6 +542,11 @@ def test_add_bot(self):
             event['event']
         )
 
+        users_result = self.client.get('/json/users')
+        members = ujson.loads(users_result.content)['members']
+        bot = filter(lambda m: m['email'] == 'hambot-bot@zulip.com', members)[0]
```

Comment:
The mypy issue is that `filter` is guaranteed to return an `Iterator`, not a list, so if you want to index it, you need to wrap a `list()` around it.  (This might be a thing where the code works in Python 2 but not Python 3)